stripe/src/main/java/com/stripe/android/paymentsheet/model/ConfirmStripeIntentParamsFactory.kt

```diff
@@ -0,0 +1,15 @@
+package com.stripe.android.paymentsheet.model
+
+import com.stripe.android.model.ConfirmPaymentIntentParams
+import com.stripe.android.model.ConfirmSetupIntentParams
+import com.stripe.android.model.ConfirmStripeIntentParams
+
+/**
+ * Factory interface for creating [ConfirmPaymentIntentParams] or [ConfirmSetupIntentParams].
+ */
+internal interface ConfirmStripeIntentParamsFactory<T : ConfirmStripeIntentParams> {
```

Comment:
Alternatively this could be a sealed class. (don't have a strong preference though)

Reply:
Yeah, that works. I'm still used to thinking Java.